course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+        });
+
+        it('creates appropriate person lookups (id or email)', function() {
+            var testCases = [
+                // [searchTerms, expectedParams]
+                [['user@example.edu'], [{email_address: 'user@example.edu'}]],
+                [['12345678'], [{univ_id: '12345678'}]],
+                [['user@example.edu', '12345678'],
+                 [{email_address: 'user@example.edu'}, {univ_id: '12345678'}]],
+            ];
+
+            spyOn(angularDRF, 'get').and.returnValue($q.resolve());
+
+            testCases.forEach(function (testCase) {
+                var searchTerms = testCase[0];
+                var expectedParams = testCase[1];
+
+                var promiseList = scope.lookupPeople(searchTerms);
+                expect(promiseList.length).toEqual(searchTerms.length);
+                expect(angularDRF.get).toHaveBeenCalledTimes(searchTerms.length);
+                for (var i=0; i<expectedParams.length; i++) {
+                    var args = angularDRF.get.calls.argsFor(i);
+                    expect(args[1].params).toEqual(expectedParams[i]);
+                }
+                angularDRF.get.calls.reset();
+            });
+        });
     });
     describe('showAddNewMemberResults', function() {
-        it('reloads datatable only if membership changed');
-        it('updates success message based on success/failure counts');
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope});
+            // handle the course instance get from setTitle, so we can always
+            // assert at the end of a test that there's no pending http calls.
+            $httpBackend.expectGET(courseInstanceURL).respond(200, '');
+            $httpBackend.flush(1);
+        });
+
+        it('reloads datatable only if membership changed', function() {
+            scope.tracking.successes = 1;
+            scope.tracking.total = 2;
+            scope.tracking.failure = 1;
+
+            // mock out the datatable so we can verify that it gets reloaded
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            //invoke the method
+            scope.showAddNewMemberResults();
+            expect(scope.dtInstance.reloadData.calls.count()).toEqual(1);
+
+        });
+
+        it('doesnt reload datatable membership is unchanged', function() {
+            scope.tracking.successes = 0;
+            scope.tracking.total = 2;
+            scope.tracking.failure = 2;
+
+            // mock out the datatable so we can verify that it gets reloaded
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            //invoke the method
+            scope.showAddNewMemberResults();
+
+            expect(scope.dtInstance.reloadData).not.toHaveBeenCalled();
+        });
+
+        it('updates success message based on success/failure counts', function(){
```

Comment:
Seems like these three could be collapsed in the `testCases.forEach()` pattern used elsewhere. But it's fine to leave them more explicit like this.